### PR TITLE
Allowing stake to be added after withdraw or after slash

### LINKED ISSUE #181: If the current stake is zero, staker should not be able to stake anymore
There are a variety of ways current stake could become zero. (penalty, inactive penalties, withdraw, etc.).

We need to write tests to verify this is expected behaviour.

### PR DESCRIPTION
**Pull request by GauravJain9**
Fixes #181 
Added test case where the staker withdraws and stakes again and then the stake should increase and then again slashing the staker and then making the staker stake again and again the stake should increase.

### CODE REVIEW

**Comment by SkandaBhat** [comment]
Please resolve conflicts @GauravJain9

File: test/StakeManager.js [hunk 1]
```diff
@@ -207,7 +206,7 @@ describe('StakeManager', function () {
       const stake = tokenAmount('20000');
       const stakerIdAcc1 = await stakeManager.stakerIds(signers[1].address);
       const stakeBeforeAcc1 = (await stakeManager.stakers(stakerIdAcc1)).stake;
-      await schellingCoin.connect(signers[1]).approve(stakeManager.address, stake);
+      await Coin.connect(signers[1]).approve(stakeManager.address, stake);
```
Comment (hrishikeshio) [inline test/StakeManager.js:209]: ?

Comment (GauravJain9) [reply test/StakeManager.js:209]: This I have changed last test case is not passing fixing it will commit soon.

Comment (GauravJain9) [reply test/StakeManager.js:209]: All errors will be fixed @hrishikeshio

**Comment by SkandaBhat** [comment]
This PR has too many changes. Please create a new one with clean commits and changes. Closing this.

**Pull request closed without merging**